src/TurbulenceClosures/turbulence_closure_diagnostics.jl

```diff
@@ -18,55 +18,64 @@ end
 min_Δxyz(grid, ::VerticalFormulation) = minimum_zspacing(grid, Center(), Center(), Center())
 
 
-cell_diffusion_timescale(model) = cell_diffusion_timescale(model.closure, model.diffusivity_fields, model.grid)
-cell_diffusion_timescale(::Nothing, diffusivities, grid) = Inf
+cell_diffusion_timescale(model) = cell_diffusion_timescale(model.closure, model.diffusivity_fields,
+                                                           model.grid, model.clock, fields(model))
+cell_diffusion_timescale(::Nothing, diffusivities, grid, clock, fields) = Inf
 
-maximum_numeric_diffusivity(κ::Number) = κ
-maximum_numeric_diffusivity(κ::FunctionField) = maximum(κ)
-maximum_numeric_diffusivity(κ_tuple::NamedTuple) = maximum(maximum_numeric_diffusivity(κ) for κ in κ_tuple)
-maximum_numeric_diffusivity(κ::NamedTuple{()}) = 0 # tracers=nothing means empty diffusivity tuples
-maximum_numeric_diffusivity(::Nothing) = 0
+maximum_numeric_diffusivity(κ::Number, grid, clock, fields) = κ
+maximum_numeric_diffusivity(κ::FunctionField, grid, clock, fields) = maximum(κ)
+maximum_numeric_diffusivity(κ_tuple::NamedTuple, grid, clock, fields) = maximum(maximum_numeric_diffusivity(κ, grid, clock, fields) for κ in κ_tuple)
+maximum_numeric_diffusivity(κ::NamedTuple{()}, grid, clock, fields) = 0 # tracers=nothing means empty diffusivity tuples
+maximum_numeric_diffusivity(::Nothing, grid, clock, fields) = 0
 
-# As the name suggests, we give up in the case of a function diffusivity
-maximum_numeric_diffusivity(κ::Function) = 0
+const FunctionDiffusivity = Union{<:DiscreteDiffusionFunction, <:Function}
 
-function cell_diffusion_timescale(closure::ScalarDiffusivity{TD, Dir}, diffusivities, grid) where {TD, Dir}
+function maximum_numeric_diffusivity(κ::FunctionDiffusivity, grid, clock, fields; ν = false)
+
+    location = (Center(), Center(), Center())
+    diffusivity = ν ?  νᶜᶜᶜ : κᶜᶜᶜ
+    diffusivity_kfo = KernelFunctionOperation{Center(), Center(), Center()}(diffusivity, grid, location, κ, clock, fields)
+    return maximum(diffusivity_kfo)
+end
```

Comment:
```suggestion
function maximum_numeric_diffusivity(κ::FunctionDiffusivity, grid, clock, fields)

    location = (Center(), Center(), Center())
    diffusivity_kfo = KernelFunctionOperation{Center(), Center(), Center()}(κᶜᶜᶜ, grid, location, κ, clock, fields)
    return maximum(diffusivity_kfo)
end
```
The two function signature are the same, so I do not think you need this keyword argument